Add special compat methods for EditorFileDialog enum functions

Bromeon opened the pull request:
While the general derived-to-base compat mechanism works, there seems to be a problem when enum types are also moved between classes. This commit restores the original enums as dummies in the derived class, to restore compatibility.

Fixes compatibility breakage introduced by #111212.

---

Earlier implementation via compat methods: https://github.com/godotengine/godot/commit/25f21edfc4109d9f82294d554a0daaee44f7096a
Current approach uses the hardcoded special mapping, due to problem with C# codegen.

Bromeon commented:
I'm not quite sure about the [C# errors](https://github.com/godotengine/godot/actions/runs/19906651560/job/57064552192?pr=113524#step:17:89):
```
Error: /home/runner/work/godot/godot/modules/mono/glue/GodotSharp/GodotSharpEditor/Generated/GodotObjects/EditorFileDialog.cs(101,16):
error CS0108: 'EditorFileDialog.GetFileMode()' hides inherited member 'FileDialog.GetFileMode()'.
Use the new keyword if hiding was intended. 
[/home/runner/work/godot/godot/modules/mono/glue/GodotSharp/GodotSharpEditor/GodotSharpEditor.csproj]
```

Do compatibility methods clash with regular ones?

dsnopek commented on editor/gui/editor_file_dialog.h:
```diff
@@ -52,8 +52,29 @@ class EditorFileDialog : public FileDialog {
+	// These enums need to exist with their original names, to be registered with backwards-compatible hashes for GDExtension.
+	enum Access {};
+	enum DisplayMode {};
+	enum FileMode {};
```
Does making these aliases via `using` work? Something like:

```suggestion
	using Access = FileDialog::Access;
	using DisplayMode = FileDialog::DisplayMode;
	using FileMode = FileDialog::FileMode;
```

dsnopek replied:
Ah, ok. It doesn't really matter in this case, but it would look nicer if it was easy :-)

Bromeon committed:
Add special compat methods for EditorFileDialog enum fns

Fixes broken binary compatibility in GDExtension, introduced
by moving methods from EditorFileDialog to its base class
FileDialog.

While the general derived-to-base compat mechanism works,
the function hashes are changed when enum types are also moved
between classes.

An earlier attempt restored original enums in EditorFileDialog
and added regular compatibility methods, however this caused
shadowed symbols in C# bindings. Instead, hash mappings are
now directly hardcoded as "special compat hashes".

dsnopek reviewed (approved):
Thanks!

I manually double checked the hash numbers, and they look good to me.

Assuming you've tested this with godot-rust (using the mode where it loads all methods) and this worked and it didn't turn up any other instances of missing methods - then I think we're good :-)
